File: SecondYear/anshika1812_AnshikaChhabra_2226CSE1186_2/Week_3/Merge two sorted arrays without using extra space.c
#include<stdio.h>

int main(){
    int arr1[100],arr2[100],i,j,k,n1,n2,end1;
    printf("Enter no. of elements in first array: ");
    scanf("%d",&n1);
    printf("Enter elements of first sorted array:\n");
    for(i=0;i<n1;i++){
        scanf("%d",&arr1[i]);
    }
    printf("Enter no. of elements in second array: ");
    scanf("%d",&n2);
    printf("Enter elements of second sorted array:\n");
    for(i=0;i<n2;i++){
        scanf("%d",&arr2[i]);
    }
    for(i=0;i<n2;i++){
        end1=arr1[n1-1];
        for(j=0;j<n1;j++){
            if(arr1[j]>arr2[0]){
                for(k=n1-2;k>=j;k--){
                    arr1[k+1]=arr1[k];
                }
                arr1[j]=arr2[0];
                break;
            }
        }
        for(j=n2-1;j>=0;j--){
            if(end1>arr2[j]){
                for(k=1;k<=j;k++){
                    arr2[k-1]=arr2[k];
                }
                arr2[j]=end1;
                break;
            }
        }
    }
    for(i=0;i<n1;i++){
        printf("%d ",arr1[i]);
    }
    for(i=0;i<n2;i++){
        printf("%d ",arr2[i]);
    }
    return 0;
}

File: SecondYear/praveer7398_PraveerNandan_2226ce1011_2/Week_3/Merge_two_sorted_arrays_without_using_Extra_space.java
import java.util.Scanner;

public class Merge_two_sorted_arrays_without_using_Extra_space {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter the size of the first array (arr1): ");
        int n = scanner.nextInt();
        long[] arr1 = new long[n];
        System.out.println("Enter the elements of the first array in non-decreasing order:");
        for (int i = 0; i < n; i++) {
            arr1[i] = scanner.nextLong();
        }

        System.out.print("Enter the size of the second array (arr2): ");
        int m = scanner.nextInt();
        long[] arr2 = new long[m];
        System.out.println("Enter the elements of the second array in non-decreasing order:");
        for (int i = 0; i < m; i++) {
            arr2[i] = scanner.nextLong();
        }
        scanner.close();

        mergeAndSort(arr1, arr2, n, m);

        for (long num : arr1) {
            System.out.print(num + " ");
        }
        System.out.println();

        for (long num : arr2) {
            System.out.print(num + " ");
        }
        System.out.println();
    }

    static void mergeAndSort(long[] arr1, long[] arr2, int n, int m) {
        int i = n - 1;
        int j = 0;

        while (i >= 0 && j < m) {
            if (arr1[i] >= arr2[j]) {
                long temp = arr1[i];
                arr1[i] = arr2[j];
                arr2[j] = temp;
                i--;
                j++;
            } else {
                break;
            }
        }

        for (int k = 0; k < n - 1; k++) {
            if (arr1[k] > arr1[k + 1]) {
                long temp = arr1[k];
                arr1[k] = arr1[k + 1];
                arr1[k + 1] = temp;
                k = -1;
            }
        }

        for (int k = 0; k < m - 1; k++) {
            if (arr2[k] > arr2[k + 1]) {
                long temp = arr2[k];
                arr2[k] = arr2[k + 1];
                arr2[k + 1] = temp;
                k = -1; 
            }
        }
    }
}
